tycho-compiler-plugin/src/main/java/org/eclipse/tycho/compiler/AbstractOsgiCompilerMojo.java

```diff
@@ -573,6 +573,7 @@ public List<String> getClasspathElements() throws MojoExecutionException {
         final List<String> classpath = new ArrayList<>();
```

Comment:
@laeubi This class `AbstractOsgiCompilerMojo` does not contain any JavaDoc on the class declaration itself. It makes me wonder if it is even supposed to work with javac, given that the plain Java compiler does not have any understanding of OSGI semantics.

Reply:
Compilation actually do not require much of OSGi, ECJ supports "restrictions" on classpath entries but `javac` not as far as I know so thats the main difference here, also the target and source levels are derived from the manifest if possible.